general_ledger/general_ledger.py

```diff
@@ -170,13 +206,16 @@ def _get_key_id(currentKey):
             key = currentKey[0].id
             return key
 
-        fiscalyear = FiscalYear(data['fiscalyear'])
+        fiscalyear = (FiscalYear(data['fiscalyear']) if data.get('fiscalyear')
+            else None)
         start_period = None
         if data['start_period']:
             start_period = Period(data['start_period'])
         end_period = None
         if data['end_period']:
             end_period = Period(data['end_period'])
+        start_date = data['start_date'] if data.get('start_date') else None
+        end_date = data['end_date'] if data .get('end_date') else None
```

Comment:
Colud be:
end_date = data.get('start_date', None)
And we solve the space between "data" and ".get"